lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -704,7 +724,126 @@ getFinalSchedule(scf::ForOp &forOp, int numStages) {
   return fSchedule;
 }
 
-// Convert load ops into their asyn version and apply multi-buffering based on
+LogicalResult
+allocTMABuffers(scf::ForOp forOp,
+                llvm::MapVector<Operation *, Value> &tmaBufferMapping,
+                int numBuffers) {
+  IRRewriter rewriter(forOp);
+
+  // Create a multi-buffered allocation for each MakeTensorDescOp call in the
+  // loop
+  forOp.walk([&](tt::MakeTensorDescOp op) {
+    // TODO peter: walk to loop yield to find the init value if this is a
+    // loop-carried value. That would save us from allocating another buffer
+    // just for the init value
```

Comment:
why do we allocate an extra buffer for init value in this case? Wouldn't the buffer be allocated only at the place where we create a descriptor?

Reply:
The initial loop value will be a value created outside the loop, most likely by a call to `tt.make_tensor_descriptor`. Since this is only replacing descriptor creation inside the loop, it will be missed and lowered by the fallback lowering instead.

Reply:
ah interesting, yes the way the original loop is written is a bit weird as the first `tt. make_tensor_descriptor` is outside the loop. I wonder if we should write it differently to make it more friendly to the pipeliner even if that means having a separate `if` block originally. But that's a detail probably not worth looking at now.